tt_eager/tt_dnn/op_library/groupnorm/groupnorm_op.hpp

```diff
@@ -38,4 +38,45 @@ Tensor groupnorm(
     const MemoryConfig& mem_config = operation::DEFAULT_OUTPUT_MEMORY_CONFIG);
 
 }  // namespace tt_metal
+
+
+namespace operations {
+
+    using namespace tt_metal;
+namespace primary {
+struct GroupNormShardedMultiCoreProgramConfig {
+    CoreCoord compute_with_storage_grid_size;
+    MathFidelity math_fidelity;
+    DataType im_data_format;
+    DataType out_data_format;
+    bool inplace;
+
+    tt::stl::reflection::Attributes attributes() const;
+};
+
+struct GroupNorm {
+    float eps;
+    uint32_t num_groups;
+    uint32_t batch;
+    MemoryConfig output_mem_config;
+    tt::operations::primary::GroupNormShardedMultiCoreProgramConfig program_config;
+
+    void validate(const std::vector<Tensor> &input_tensors, const std::vector<std::optional<const Tensor>>& optional_input_tensors) const;
+    std::vector<Shape> compute_output_shapes(const std::vector<Tensor> &input_tensors) const;
+    std::vector<Tensor> create_output_tensors(const std::vector<Tensor> &input_tensors) const;
+    operation::ProgramWithCallbacks create_program(
+        const std::vector<Tensor>& input_tensors,
+        const std::vector<std::optional<const Tensor>>& optional_input_tensors,
+        std::vector<Tensor> &output_tensors
+    ) const;
+    tt::stl::reflection::Attributes attributes() const;
+};
+
+inline Tensor groupnorm(const Tensor &a, const uint32_t num_groups, uint32_t batch, float eps, std::optional<const Tensor> gamma = std::nullopt, std::optional<const Tensor> beta = std::nullopt, const MemoryConfig& output_mem_config = operation::DEFAULT_OUTPUT_MEMORY_CONFIG, const GroupNormShardedMultiCoreProgramConfig& program_config = GroupNormShardedMultiCoreProgramConfig{}) {
```

Comment:
Does `batch` needs to be passed in? Can we not infer it from the shape?

Reply:
I think the conversation op fuse the batch to shape[2]. so the input tensor has shape (1, 1, num_batches * W * H, C). @mywoodstock pls correct me if I'm wrong here

Reply:
removed the batch arg